### PR TITLE
Initial pass at fixing non-symlink redirects

### PR DESCRIPTION
**Pull request by dobooth**
## This PR is a:

Bug fix

## Summary

When this pull request is merged, it will:
- Fix multiple redirects in files that are not symlinks.
- Try to get the least number of possible redirects.
- Leave 2.0 redirects as they will be dealt within en masse when we get rid of 2.0 docs.
- Try to prevent redirects from jumping versions.

### CODE REVIEW

File: guides/v2.1/cloud/release-notes/CloudReleaseNotes101.4.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ menu_node:
 level3_menu_node: level3child
 level3_subgroup: mccloud-relnotes
 redirect_from:
-  - /guides/v2.2/cloud/release-notes/CloudReleaseNotes101.4.html
+  - /guides/v2.1/cloud/release-notes/CloudReleaseNotes101.4.html
```
Comment (dshevtsov) [inline guides/v2.1/cloud/release-notes/CloudReleaseNotes101.4.md]: this redirects the file to itself

File: guides/v2.1/cloud/release-notes/CloudReleaseNotes101.6.md [hunk 2]
```diff
@@ -8,7 +8,7 @@ menu_node:
 level3_menu_node: level3child
 level3_subgroup: mccloud-relnotes
 redirect_from:
-  - /guides/v2.2/cloud/release-notes/CloudReleaseNotes101.6.html
+  - /guides/v2.1/cloud/release-notes/CloudReleaseNotes101.6.html
```
Comment (dshevtsov) [inline guides/v2.1/cloud/release-notes/CloudReleaseNotes101.6.md]: this redirects the file to itself

File: guides/v2.1/cloud/release-notes/CloudReleaseNotes101.7.md [hunk 3]
```diff
@@ -9,7 +9,7 @@ level3_menu_node: level3child
 level3_subgroup: mccloud-relnotes
 redirect_from:
   - /guides/v2.0/cloud/release-notes/CloudReleaseNotes101.7.html
-  - /guides/v2.2/cloud/release-notes/CloudReleaseNotes101.7.html
+  - /guides/v2.1/cloud/release-notes/CloudReleaseNotes101.7.html
```
Comment (dshevtsov) [inline guides/v2.1/cloud/release-notes/CloudReleaseNotes101.7.md]: this redirects the file to itself

File: guides/v2.1/cloud/release-notes/CloudReleaseNotes101.4.md (hunk 1 above)
Comment (dobooth) [reply guides/v2.1/cloud/release-notes/CloudReleaseNotes101.4.md]: Thanks.

File: guides/v2.1/cloud/release-notes/CloudReleaseNotes101.6.md (hunk 2 above)
Comment (dobooth) [reply guides/v2.1/cloud/release-notes/CloudReleaseNotes101.6.md]: I'll fix them all.

File: guides/v2.1/cloud/release-notes/CloudReleaseNotes101.7.md (hunk 3 above)
Comment (dobooth) [reply guides/v2.1/cloud/release-notes/CloudReleaseNotes101.7.md]: Fixed.

**Comment by keharper** [comment]
It's safe to delete the redirect statement from these topics

guides/v2.3/rest/performing-searches.md 
guides/v2.3/rest/retrieve-filtered-responses.md

**Comment by dobooth** [comment]
jekyll doctor count down from 427? to 387.

**Comment by dobooth** [comment]
Other writers reviewing here need only review those links in their work areas.

**Comment by dobooth** [comment]
running tests

**Pull request merged**